### PR TITLE
Building Backend Plugins and Modules

### PR DESCRIPTION
**Pull request by freben**

### CODE REVIEW

File: docs/backend-system/building-plugins-and-modules/01-index.md [hunk 1]
```diff
@@ -9,4 +9,236 @@ description: Building backend plugins and modules using the new backend system
...
+        logger: coreServices.logger,
+        httpRouter: coreServices.httpRouter,
+      },
+      async init({
+        // Requested service instances get injected as per above
+        logger,
+        httpRouter,
+      }) {
+        // Perform your initialization and access the services as needed
+        const router = createExampleRouter(logger);
```
Comment (benjdlambert) [inline docs/backend-system/building-plugins-and-modules/01-index.md:45]:
```suggestion
        const example = createExampleRouter(logger);
```

File: docs/backend-system/building-plugins-and-modules/01-index.md [hunk 2]
```diff
@@ -9,4 +9,236 @@ description: Building backend plugins and modules using the new backend system
...
+### Database Access
+
+The same applies for modules that perform their own migrations and interact with
+the database. They will run on the same logical database instance as the target
+plugin, so care must be taken to choose table names that do not risk colliding
+with those of the plugin. A recommended naming pattern is `<module name>__<table
+name>`, for example the `@backstage/backend-tasks` package creates tables named
+`backstage_backend_tasks__<table>`. If you use the default Knex migration
+facilities, you will also want to make sure that it uses a similarly prefixed
+lock file for its internal bookkeeping needs, so it does not collide with the
```
Comment (benjdlambert) [inline docs/backend-system/building-plugins-and-modules/01-index.md:144]: `migration state table` instead of `lock file`?

**Review by benjdlambert — APPROVED** [review]
Nice work! Just two small things, but unblocking anyways!

File: docs/backend-system/building-plugins-and-modules/01-index.md (hunk 2 above)
Comment (freben) [reply docs/backend-system/building-plugins-and-modules/01-index.md]: 👍 yeah that reads better, there's a lock table indeed but your wording makes more sense

**Pull request merged**